SpectatorView/Compositor/CompositorDLL/DeckLinkDevice.cpp

```diff
@@ -142,34 +142,40 @@ ULONG STDMETHODCALLTYPE DeckLinkDevice::Release(void)
 
 bool DeckLinkDevice::Init(ID3D11ShaderResourceView* colorSRV, ID3D11Texture2D* outputTexture, bool useCPU, bool passthroughOutput)
 {
-    IDeckLinkAttributes*            deckLinkAttributes = NULL;
-    IDeckLinkDisplayModeIterator*   displayModeIterator = NULL;
-    IDeckLinkDisplayMode*           displayMode = NULL;
-    BSTR                            deviceNameBSTR = NULL;
-
-    ZeroMemory(rawBuffer, FRAME_BUFSIZE_RAW);
-    ZeroMemory(cachedBuffer, FRAME_BUFSIZE);
-    ZeroMemory(latestBuffer, FRAME_BUFSIZE);
-    ZeroMemory(outputBuffer, FRAME_BUFSIZE);
-    ZeroMemory(outputBufferRaw, FRAME_BUFSIZE_RAW);
-
-    _useCPU = useCPU;
-    _passthroughOutput = passthroughOutput;
-
-    _colorSRV = colorSRV;
-    _outputTexture = outputTexture;
-
-    if (colorSRV != nullptr)
-    {
-        colorSRV->GetDevice(&device);
-    }
-
-    // Get input interface
-    if (m_deckLink->QueryInterface(IID_IDeckLinkInput, (void**) &m_deckLinkInput) != S_OK)
-        return false;
-
-    if (m_deckLink->QueryInterface(IID_IDeckLinkOutput, (void**)&m_deckLinkOutput) != S_OK)
-        return false;
+	IDeckLinkAttributes*            deckLinkAttributes = NULL;
+	IDeckLinkDisplayModeIterator*   displayModeIterator = NULL;
+	IDeckLinkDisplayMode*           displayMode = NULL;
+	BSTR                            deviceNameBSTR = NULL;
+
+	ZeroMemory(rawBuffer, FRAME_BUFSIZE_RAW);
+	ZeroMemory(cachedBuffer, FRAME_BUFSIZE);
+	ZeroMemory(latestBuffer, FRAME_BUFSIZE);
+	ZeroMemory(outputBuffer, FRAME_BUFSIZE);
+	ZeroMemory(outputBufferRaw, FRAME_BUFSIZE_RAW);
+
+	_useCPU = useCPU;
+	_passthroughOutput = passthroughOutput;
+
+	_colorSRV = colorSRV;
+	_outputTexture = outputTexture;
+
+	if (colorSRV != nullptr)
+	{
+		colorSRV->GetDevice(&device);
+	}
+
+	// Get input interface
+	if (m_deckLink->QueryInterface(IID_IDeckLinkInput, (void**)&m_deckLinkInput) != S_OK)
+		return false;
+
+	if (m_deckLink->QueryInterface(IID_IDeckLinkOutput, (void**)&m_deckLinkOutput) != S_OK)
+	{
+		supportsOutput = false;
+		// Returning false here prevent the rest of the init function executing
+		// But we can still continue without an output interface
+		// return false;

```

Comment:
Great stuff. I'll make the alterations tomorrow morning. 
Good to know it didn't break cards with output!